src/fu-util.c

```diff
@@ -2466,10 +2431,19 @@ fu_util_update(FuUtilPrivate *priv, gchar **values, GError **error)
 		return FALSE;
 	}
 
-	if (g_strv_length(values) == 0)
-		return fu_util_update_all(priv, error);
-	if (g_strv_length(values) == 1)
-		return fu_util_update_by_id(priv, values[0], error);
+	/* DEVICE-ID and GUID are acceptable args to update */
+	for (guint idx = 0; idx < g_strv_length(values); idx++) {
+		if (!fwupd_guid_is_valid(values[idx]) && !fwupd_device_id_is_valid(values[idx])) {
+			g_set_error_literal(error,
+					    FWUPD_ERROR,
+					    FWUPD_ERROR_INVALID_ARGS,
+					    "Invalid arguments");
```

Comment:
I think that now that multiple arguments are possible to specify, it's best to output _which_ argument caused the failure to make this easier for a user to use.